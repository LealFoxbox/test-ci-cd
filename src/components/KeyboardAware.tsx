import React, { ReactNode } from 'react';
import { Platform } from 'react-native';
import { KeyboardAwareScrollView, KeyboardAwareScrollViewProps } from 'react-native-keyboard-aware-scroll-view';

const isIOS = Platform.OS === 'ios';

interface ScrollViewProps extends KeyboardAwareScrollViewProps {
  children: ReactNode;
  ref: any;
}

type Ref =
  | ((instance: KeyboardAwareScrollView | null) => void)
  | React.MutableRefObject<KeyboardAwareScrollView | null>
  | null;

export const ScrollView = React.forwardRef(({ children, style, ...props }: ScrollViewProps, ref: Ref) => (
  <KeyboardAwareScrollView
    style={[{ flex: 1, width: '100%' }, style]}
    enableOnAndroid
    extraScrollHeight={30}
    keyboardShouldPersistTaps="handled"
    enableAutomaticScroll={isIOS}
    {...props}
    ref={ref}
  >
    {children}
  </KeyboardAwareScrollView>
));
