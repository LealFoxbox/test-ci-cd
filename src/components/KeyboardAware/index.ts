export * from './KeyboardAware';
// export { default } from './KeyboardAware';
