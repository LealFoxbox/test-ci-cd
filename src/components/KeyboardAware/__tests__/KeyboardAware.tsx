import React from 'react';
import { View } from 'react-native';
import { render } from '@testing-library/react-native';

import { ScrollView } from '../KeyboardAware';

describe('KeyboardAware.ScrollView', () => {
  it('should render', () => {
    const result = render(
      <ScrollView>
        <View />
      </ScrollView>,
    );

    expect(result).toMatchSnapshot();
  });
});
