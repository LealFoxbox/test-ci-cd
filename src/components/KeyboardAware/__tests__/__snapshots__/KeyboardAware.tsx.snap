// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyboardAware.ScrollView should render 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutomaticScroll={true}
  enableOnAndroid={true}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={30}
  getScrollResponder={[Function]}
  handleOnScroll={[Function]}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  keyboardShouldPersistTaps="handled"
  keyboardSpace={0}
  onScroll={[Function]}
  resetKeyboardSpace={[Function]}
  scrollEventThrottle={1}
  scrollForExtraHeightOnAndroid={[Function]}
  scrollIntoView={[Function]}
  scrollToEnd={[Function]}
  scrollToFocusedInput={[Function]}
  scrollToPosition={[Function]}
  showsVerticalScrollIndicator={true}
  style={
    Array [
      Object {
        "flex": 1,
        "width": "100%",
      },
      undefined,
    ]
  }
  update={[Function]}
  viewIsInsideTabBar={false}
>
  <View>
    <View />
  </View>
</RCTScrollView>
`;
